### PR TITLE
ENH: stats.kstests: add axis / nan_policy / keepdims support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Toward gh-14651

#### What does this implement/fix?
Adds axis / nan_policy / keepdims support to `scipy.stats.kstest` and related functions.

### CODE REVIEW

File: scipy/stats/tests/test_axis_nan_policy.py [hunk 1]
```diff
@@ -65,6 +66,14 @@ def ttest_ci(*args, **kwargs):
     (stats.mode, tuple(), dict(), 1, 2, True, lambda x: (x.mode, x.count)),
     (stats.differential_entropy, tuple(), dict(), 1, 1, False, lambda x: (x,)),
     (stats.variation, tuple(), dict(), 1, 1, False, lambda x: (x,)),
+    (stats.ks_1samp, (norm().cdf,), dict(), 1, 4, False,
```
Comment (tirthasheshpatel) [inline scipy/stats/tests/test_axis_nan_policy.py:69]: nit: `stats.norm.cdf` to save an import.

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -8392,36 +8405,35 @@ def ks_1samp(x, cdf, args=(), alternative='two-sided', method='auto'):
         alternative.lower()[0], alternative)
     if alternative not in ['two-sided', 'greater', 'less']:
         raise ValueError("Unexpected alternative %s" % alternative)
-    if np.ma.is_masked(x):
-        x = x.compressed()
 
     N = len(x)
     x = np.sort(x)
     cdfvals = cdf(x, *args)
+    np_one = np.int8(1)
```
[the remark this answers is not in the document]
Comment (mdhaber) [reply scipy/stats/_stats_py.py:8412]: Presumably,but it's been so long I don't remember what it was.

[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply scipy/stats/_stats_py.py:8412]: Looks like the tests check if the received outputs have a `np.number` data type. The only difference I could find was that the `statistic_sign` now is a `np.float64` array for multi-dimensional inputs while, for 1-D inputs and `axis=None`, we return an `np.int8` output. I think that's OK to do; NumPy's operations should handle dtype promotion correctly if the user chooses to use the sign for any computations.

**Review by tirthasheshpatel — APPROVED** [review]
Everything looks good now! Thanks @mdhaber!

**Comment by mdhaber** [comment]
Thanks @tirthasheshpatel. I'll open up another to help us catch up. Please also take a look at the spreadsheet - I think `f_oneway`, `alexandergovern`, `bws_test`, and `combine_pvalues` are about all that's left of the easy stuff.

**Pull request merged**